src/main/java/mekanism/common/content/gear/mekasuit/ModuleMekaSuit.java

```diff
@@ -45,15 +44,21 @@
 
         @Override
         public void tickServer(PlayerEntity player) {
-            if (player.func_233571_b_(FluidTags.WATER) > 1.8) {
+            if (player.func_233571_b_(FluidTags.WATER) > 1.6) {
                 FloatingLong usage = MekanismConfig.general.FROM_H2.get().multiply(2);
                 long maxRate = Math.min(getMaxRate(), getContainerEnergy().divide(usage).intValue());
                 long hydrogenUsed = 0;
                 GasStack hydrogenStack = new GasStack(MekanismGases.HYDROGEN.get(), maxRate * 2);
                 ItemStack chestStack = player.getItemStackFromSlot(EquipmentSlotType.CHEST);
-                Optional<IGasHandler> capability = MekanismUtils.toOptional(chestStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
-                if (Modules.load(chestStack, Modules.JETPACK_UNIT) != null && capability.isPresent()) {
-                    hydrogenUsed = maxRate * 2 - capability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
+                Optional<IGasHandler> chestCapability = MekanismUtils.toOptional(chestStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
+                if (chestCapability.isPresent()) {
+                    hydrogenUsed = maxRate * 2 - chestCapability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
+                }
+                hydrogenStack.setAmount(maxRate * 2 - hydrogenUsed);
+                ItemStack handStack = player.getItemStackFromSlot(EquipmentSlotType.MAINHAND);
+                Optional<IGasHandler> handCapability = MekanismUtils.toOptional(handStack.getCapability(Capabilities.GAS_HANDLER_CAPABILITY));
+                if (handCapability.isPresent()) {
+                	hydrogenUsed -= handCapability.get().insertChemical(hydrogenStack, Action.EXECUTE).getAmount();
```

Comment:
Given that `hydrogenUsed` is total hydrogen used, and we adjust the hydrogen stack to be what the amount we have is, and that `insertChemical` returns the remainder, shouldn't this line be basically identical to the one for the chestCapability. That way it is `max - remainingAmount`

Reply:
insertChemical returns the remainder? I just assumed it returns the inserted amount.
In that case, yes, it should be max - remainder.

Reply:
Ya, our handlers (including internal implementation of fluid handling) are closer in behavior to forge's item handlers than forge's fluid handlers in that we have insert return the remainder (as it helps cut down on a decent bit of code across the board for doing repetitive functions)

Reply:
done.